Session_5/module_5.py

```diff
@@ -21,20 +24,94 @@
 
 def task_1():
     seed(1)
-    pass
+    # I open names file and split into lines and then lowercasing and sorting
+    with open(PATH_TO_NAMES, 'r', encoding='utf-8') as names_file:
+        names = names_file.read().splitlines()
+        names = [name.lower() for name in names]
+        names.sort()
+
+    # reading surnames and splitting into lines
+    with open(PATH_TO_SURNAMES, 'r', encoding='utf-8') as surnames_file:
+        surnames = surnames_file.read().splitlines()
+
+    # assigning random surnames to names
+    sorted_names_and_surnames = [f"{name} {choice(surnames)}" for name in names]
+
+    with open(PATH_TO_OUTPUT, 'w', encoding='utf-8') as output_file:
+        for entry in sorted_names_and_surnames:
+            output_file.write(f"{entry}\n")
 
 
 def task_2(top_k: int):
-    pass
+    # reading stop words
+    with open(PATH_TO_STOP_WORDS, 'r', encoding='utf-8') as stop_words_file:
+        stop_words = set(stop_words_file.read().splitlines())
+
+    # reading random_text file
+    with open(PATH_TO_TEXT, 'r', encoding='utf-8') as text_file:
+        text = text_file.read().lower()
+
+    # Extract words, keeping only alphabetic tokens
+    words = ''.join(char if char.isalpha() else ' ' for char in text).split()
+
+    # removing stop words from the text
+    filtered_words = [word for word in words if word not in stop_words]
+
+    # counting how many times words appears, creating empty dictionary and for every time word appears
+    # dictionary will rise word count by one, what way we will know how many times each work appears in the file
+
+    word_counts = {}
+    for word in filtered_words:
+        if word in word_counts:
+            word_counts[word] += 1
+        else:
+            word_counts[word] = 1
+
+    # sorting in reversing mode the tuple of words and counts and taking the top k
+    sorted_word_counts = sorted(word_counts.items(), key=lambda item: item[1], reverse=True)
+    top_words = sorted_word_counts[:top_k]
+
+    return top_words
 
 
 def task_3(url: str):
-    pass
+    try:
+        response = requests.get(url)
+        response.raise_for_status()
+        return response
+    except requests.exceptions.HTTPError as e:
+        raise RequestException(e)
+    except RequestException as e:
+        raise e
 
 
 def task_4(data: List[Union[int, str, float]]):
-    pass
+    sum1 = 0.0
+    for i in data:
+        try:
+            # converting into  float and adding to the sum
+            sum1 += float(i)
+        except ValueError as e:
+            # if it is impossible to convert to float raise error
+            raise TypeError(f"Cannot convert element to float: {i}") from e
```

Comment:
In this task you shouldn't raise `TypeError` yourself, instead you should try to sum up all elements as is, if you get `TypeError` process it and convert element to float to add it to sum. If you get `ValueError` reraise it or print that sum couldn't be counted.